manager/integration/tests/test_settings.py

```diff
@@ -277,48 +301,43 @@ def test_setting_toleration_extra(core_api, apps_api):  # NOQA
 
 def wait_for_toleration_update(core_api, apps_api, count,  # NOQA
                                expected_tolerations,
-                               chk_removed_tolerations=[]):
-    not_managed_apps = [
-        "csi-attacher",
-        "csi-provisioner",
-        "csi-resizer",
-        "csi-snapshotter",
-        "longhorn-csi-plugin",
+                               chk_removed_tolerations=[],
+                               backing_image=""):
+    skip_updates = [
         "longhorn-driver-deployer",
         "longhorn-manager",
         "longhorn-ui",
     ]
-    updated = False
+    ok = False
     for _ in range(RETRY_COUNTS):
         time.sleep(RETRY_INTERVAL_LONG)
 
-        updated = True
-        if not check_workload_update(core_api, apps_api, count):
-            updated = False
+        ok = False
+        if not check_longhorn_managed_workloads(core_api, apps_api, count):
             continue
 
         pod_list = core_api.list_namespaced_pod(LONGHORN_NAMESPACE).items
+
         for p in pod_list:
-            managed_by = p.metadata.labels.get('longhorn.io/managed-by', '')
-            if str(managed_by) != "longhorn-manager":
+            if backing_image != "" and \
+                    not p.metadata.name.startswith("backing-image-manager"):
                 continue
-            else:
-                app_name = str(p.metadata.labels.get('app', ''))
-                assert app_name not in not_managed_apps
-
-            if p.status.phase != "Running" \
-                or not check_tolerations_set(p.spec.tolerations,
-                                             expected_tolerations,
-                                             chk_removed_tolerations):
-                updated = False
-                break
-        if updated:
+
+            n_update = check_tolerations_set_update(p.spec.tolerations,
+                                                    expected_tolerations,
+                                                    chk_removed_tolerations)
+            app_name = str(p.metadata.labels.get('app', ''))
+            if app_name in skip_updates and n_update == 0:
+                ok = True
+            elif n_update == len(expected_tolerations):
+                ok = True
+        if ok:
             break
-    assert updated
+    assert ok
```

Comment:
Why not put this backing image manager pod checking into function `check_longhorn_managed_workloads`?

Besides, you can use the following logic to determine if a backing image manager pod **should** exist:
    1. List all BackingImages
    2. For each BackingImage, each entry in `diskStateMap` means there should be a backing image manager pod running for the disk. The key of `diskStateMap` is disk UUID. After iterating BackingImages, we can get the disks that require backing image managers.
    3. List all backing image managers, there should be a backing image manager for each entry of the disk list. And the related pod is running with correct tolerations & priority class.

Reply:
Fixed!